compiler/modules/nback/compare.bal

```diff
@@ -137,6 +156,9 @@ function buildCompare(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn i
                 if isOperandIntSubtypeArray(tc, lhs) && isOperandIntSubtypeArray(tc, rhs) {
                     buildCompareSpecializedIntList(builder, scaffold, insn, lhsValue, rhsValue);
                 }
+                else if isOperandDecimalSubtypeArray(tc, lhs) && isOperandDecimalSubtypeArray(tc, rhs) {
```

Comment:
The decimal comparison should work like string: do a buildCompareDecimal like buildCompareString. Use decimal_cmp. We don't need a specialized version of compare for arrays of decimals.

Reply:
If we replicate what we do with string here, similar to string, in a case like 
```
decimal[] x = [1.5d];
decimal[] y = [3d];
io:println(x < y); // @output true
io:println(x > y); // @output false
```
we would be using `array_generic_compare` right? In such a case `array_generic_compare` doesn't seem to give the correct result with decimal values.